tests/test_tasks.py

```diff
@@ -0,0 +1,294 @@
+        self.assertEqual(Task.query.count(), 0)
+
+    def test_put_task_with_no_status(self):
+        task = Task(title='title1', status='in_progress')
+        task.assign_user(1)
+        db.session.add(task)
+        db.session.commit()
+        res = self.client.get(self.ENDPOINT + '/1', headers=self.auth_header)
+        self.assert200(res)
+        self.assertEqual(res.json['status'], 'in_progress')
+        self.assertEqual(Task.query.get(1).status, 'in_progress')
+        res = self.client.put(self.ENDPOINT + '/1',
+                              headers=self.auth_header)
+        self.assert400(res)
+        self.assertEqual(Task.query.get(1).status, 'in_progress')
+
+    def test_put_task_data_error(self):
+        task = Task(title='title1', status='in_progress')
+        task.assign_user(1)
+        db.session.add(task)
+        db.session.commit()
+        res = self.client.get(self.ENDPOINT + '/1', headers=self.auth_header)
+        self.assert200(res)
+        self.assertEqual(res.json['status'], 'in_progress')
+        self.assertEqual(Task.query.get(1).status, 'in_progress')
+        res = self.client.put(self.ENDPOINT + '/1',
+                              headers=self.auth_header,
+                              data={'status': 'statusstatusstatus'})
+        self.assert400(res)
+        self.assertEqual(Task.query.get(1).status, 'in_progress')
+
+    def test_put_other_user_task(self):
+        task1 = Task(title='title1', status='in_progress')
+        task1.assign_user(1)
+        db.session.add(task1)
+        db.session.commit()
+        self.assertEqual(Task.query.get(1).status, 'in_progress')
+        res = self.client.get(self.ENDPOINT + '/1', headers=self.auth_header)
+        self.assert200(res)
+        user2 = User(name='user2', email='email2@em.co')
+        user2.password = hashlib.md5('1').hexdigest()
+        db.session.add(user2)
+        db.session.commit()
+        token = self.login('user2', '1').json['token']
+        res = self.client.put(self.ENDPOINT + '/1',
+                              headers={'token': token},
+                              data={'status': 'completed'})
+        self.assert403(res)
+        self.assertEqual(Task.query.get(1).status, 'in_progress')
+
+    def test_put_user_task(self):
+        task1 = Task(title='title1', status='in_progress')
+        task1.assign_user(1)
+        db.session.add(task1)
+        db.session.commit()
+        self.assertEqual(Task.query.get(1).status, 'in_progress')
+        res = self.client.get(self.ENDPOINT + '/1', headers=self.auth_header)
+        self.assert200(res)
+        res = self.client.put(self.ENDPOINT + '/1',
+                              headers=self.auth_header,
+                              data={'status': 'completed'})
+        self.assert200(res)
+        self.assertEqual(Task.query.get(1).status, 'completed')
+
+
+class TestAssignTask(BaseTestCase):
+
+    ENDPOINT = 'api/assign_task'
```

Comment:
I think we need 2 more tests:
1. test for 403
2. test for duplicated assignment